File: market/test/java/ru/yandex/market/mbo/taskqueue/config/DbConfig.java
package ru.yandex.market.mbo.taskqueue.config;

import javax.sql.DataSource;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.autoconfigure.liquibase.DataSourceClosingSpringLiquibase;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.PropertySource;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
import org.springframework.jdbc.datasource.DataSourceTransactionManager;
import org.springframework.jdbc.datasource.DriverManagerDataSource;
import org.springframework.transaction.PlatformTransactionManager;
import org.springframework.transaction.support.TransactionTemplate;

import ru.yandex.market.mboc.common.infrastructure.sql.TransactionHelper;


@Configuration
@PropertySource("classpath:db-test.properties")
public class DbConfig {

    @Value("${sql.driverName}")
    private String jdbcDriver;
    @Value("${sql.url}")
    private String url;
    @Value("${sql.userName}")
    private String username;
    @Value("${sql.password}")
    private String password;

    @Value("${sql.liquibase.tables.schema}")
    private String liquibaseTablesSchema;
    @Value("${sql.liquibase.changelog}")
    private String liquibaseChangelog;
    @Value("${sql.liquibase.enabled}")
    private boolean liquibaseEnabled;

    @Bean
    public DataSourceClosingSpringLiquibase liquibase() {
        DataSourceClosingSpringLiquibase liquibase = new DataSourceClosingSpringLiquibase();
        liquibase.setDataSource(dataSource());
        liquibase.setDefaultSchema(liquibaseTablesSchema);
        liquibase.setChangeLog("classpath:" + liquibaseChangelog);
        liquibase.setShouldRun(liquibaseEnabled);
        return liquibase;
    }

    @Bean
    public DataSource dataSource() {
        DriverManagerDataSource dataSource = new DriverManagerDataSource(url, username, password);
        dataSource.setDriverClassName(jdbcDriver);
        return dataSource;
    }

    @Bean
    public PlatformTransactionManager transactionManager() {
        return new DataSourceTransactionManager(dataSource());
    }

    @Bean
    public TransactionTemplate transactionTemplate() {
        return new TransactionTemplate(transactionManager());
    }

    @Bean
    public JdbcTemplate jdbcTemplate() {
        return new JdbcTemplate(dataSource());
    }

    @Bean
    public NamedParameterJdbcTemplate namedParameterJdbcTemplate() {
        return new NamedParameterJdbcTemplate(jdbcTemplate());
    }

    @Bean
    public TransactionHelper transactionHelper(TransactionTemplate transactionTemplate) {
        return transactionTemplate::execute;
    }
}

File: market/test/kotlin/ru/yandex/market/wms/placement/config/DbConfig.kt
package ru.yandex.market.wms.placement.config

import org.dbunit.database.DatabaseDataSourceConnection
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.autoconfigure.jdbc.DataSourceProperties
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.context.annotation.Profile
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
import ru.yandex.market.wms.common.service.DbConfigService
import ru.yandex.market.wms.common.spring.config.IntegrationTestConfig
import ru.yandex.market.wms.common.spring.utils.FileContentUtils
import ru.yandex.market.wms.placement.dao.NsqlCacheDao
import ru.yandex.market.wms.shared.libs.env.conifg.Profiles
import javax.sql.DataSource

@Configuration
@Profile(ru.yandex.market.wms.shared.libs.env.conifg.Profiles.TEST)
@SpringBootApplication(scanBasePackages = ["ru.yandex.market.wms.placement"])
class DbConfig {
    @Bean
    @Primary
    fun dataSource(): DataSource {
        val dataSource = dataSourceProperties().initializeDataSourceBuilder().build();
        val initScriptContent = FileContentUtils.getFileContent("placement-schema.sql");
        dataSource.connection.prepareStatement(initScriptContent).execute();
        return dataSource;
    }

    @Bean
    @Primary
    @ConfigurationProperties("spring.datasource-placement")
    fun dataSourceProperties() = DataSourceProperties()

    @Bean
    @Primary
    fun namedParameterJdbcTemplate() = NamedParameterJdbcTemplate(dataSource())

    @Bean
    fun placementConnection(): DatabaseDataSourceConnection {
        return IntegrationTestConfig.dbUnitDatabaseConnection("PLACEMENT", dataSource())
    }

    @Bean
    fun archiveConnection(): DatabaseDataSourceConnection {
        return IntegrationTestConfig.dbUnitDatabaseConnection("ARCHIVE", dataSource())
    }

    @Bean
    @Primary
    fun dbConfigService(nsqlCacheDao: NsqlCacheDao) = DbConfigService(nsqlCacheDao)
}
